pennylane/pauli/pauli_arithmetic.py

```diff
@@ -397,3 +398,9 @@ def simplify(self, tol=1e-8):
         for pw, coeff in items:
             if abs(coeff) <= tol:
                 del self[pw]
+
+    def split(self, max_size=20000):
+        """Splits into sub-PauliSentences of size determined by the `max_size`."""
+        it, length = iter(self), len(self)
+        for _ in range(0, length, max_size):
+            yield PauliSentence({k: self[k] for k in islice(it, max_size)})
```

Comment:
Do we need to add this to the `PauliSentence` class itself, or could we just have this functionality as a private helper function in the `tapering` page?

If this code has a very specific use case, we should make this a responsibility of the tapering module, and not the `PauliSentence` class itself. Then `tapering` can change how it does things without having to go back and change the `PauliSentence` interface.

We want to reduce coupling between objects to well-defined, minimal, public interfaces and keep implementation details private wherever possible.